lib/rucio/common/utils.py

```diff
@@ -897,39 +897,39 @@ class Color:
 
 def detect_client_location():
     """
-    Open a UDP socket to a machine on the internet, to get the local IPv4 and IPv6
-    addresses of the requesting client.
+    Client IP will be set on the server side (request.remote_addr)
 
     Try to determine the sitename automatically from common environment variables,
     in this order: SITE_NAME, ATLAS_SITE_NAME, OSG_SITE_NAME. If none of these exist
     use the fixed string 'ROAMING'.
+
+    If environment variables sets location, it uses it.
     """
 
     ip = '0.0.0.0'
```

Comment:
Thinking more about this, I think we should leave the ip detection in the dictionary. (Without any outside connection, just reading it from the interface)
This method might be used (even in the future) by other methods, like traces, and they could use the local ip. (E.g. for a trace, you might want to know which node was it actually which created the data)
For the list_replicas, we will not use it and use the rest-exposed remote address (That you already changed)

Reply:
leave socket method? I think that's just opening a possibility for future issues and inconsistencies. 
Maybe we create three functions: one simply detecting site leaving everything else None, one detecting IP using socket where local ip is OK, one detecting using external service.

Reply:
As @mlassnig  explained, since the `socket.SOCK_DGRAM` is used, I think it is fine, there is no outgoing connection for UDP. 
I would not break up the dictionary, as the method already needs to include fqdn, site, lat, long - it makes sense to leave ip and ip6 in. What we could do in the future is to have some options to the method do detect things differently.

Reply:
I know there is no outgoing connection, but most people will be unaware of issues connected to the method. 
Also I would remove ipv6 from the dictionary. It creates false sense that in case you have ipv6 only node, things will work when in fact they wont. We leave only one ip. If it is ivp4 thats great, if that's ivp6 that's ok too.
I would not break the dictionary just have different functions that fill/overwrite its components...

Reply:
pulling in @PalNilsson . I added this ipv6 field for the pilot, because he asked me to :-)

Reply:
Well, the pilot version of this function is a bit different;
[(pilot source)](https://github.com/PanDAWMS/pilot2/blob/29c4352edd9df14a480fa8b406c4a506e2313bc1/pilot/api/data.py#L372)
So if you remove the ip6 field from the dictionary, I (or better, you) will have to update the pilot code as well. From which version?

I wonder if I could just pull this function from the rucio API instead of duplicating it, which otherwise inevitably leads to situations like this. RUCIO_LAT/LONGITUDE don't appear to exist on the WNs though. I suppose you will have them set later..?

Reply:
can someone please explain to how are these two functions related? Paul, what exactly pilot uses ip and ipv6 values it obtains?

Reply:
As far as I can see Paul uses that "duplicated" `detect_client_location` to feed the `list_replica` calls the pilot makes, it is not used otherwise. To make use of RUCIO_LAT/LONG he would have to update that one as well. (Alternatively, you could also just import the one from `rucio.common.utils`)
For ip4/6 you actually don't need to change it, since for your usecase (list_replicas) the server will anyway use the incoming ip.

Coming back to `detect_client_location` itself:
I lean to merging this as well and just having `ip` (Precedence maybe ip6>ip4); Although public API, I don't think this is used by anyone, so I don't think it will break things. We should make it clear in the description how `ip` is generated and that it will simply be the device address, not location in a global sense.
As far as I can see this should work and will not break anything, as it is not used anywhere else.

Reply:
So, is there anything you want me change?

Reply:
Yes, bring the ip detection back to `detect_client_location`; Just make it one field `ip`, which gives either ip4 or ip6, if both available, ip6.

Reply:
done now.